ecommerce/products/templates/home.html

```diff
@@ -5,13 +5,19 @@ <h2>Recently added</h2>
 <ul>
     <li>
         <div>
+            <a href="{{ product.id }}">
```

Comment:
You need to update this url to be like the one below. What you're doing here is what I was describing as the "naive" approach. It won't work if you decide to change the url patterns later.